uimaj-core/src/main/java/org/apache/uima/jcas/tcas/Annotation.java

```diff
@@ -244,4 +245,39 @@ public final int compareAnnotationWithId(Annotation other, LinearTypeOrder lto)
     return Integer.compare(_id,  other._id);
   }
 
+  @Override
+  public void trim(IntPredicate aIsTrimChar) {
+    int begin = getBegin();
+    int end = getEnd();
+    String text = _casView.getDocumentText();
+      
+    // If the span is empty, there is nothing to trim
+    if (begin == end) {
+      return;
+    }
+      
+    // First we trim at the end. If a trimmed span is empty, we want to return the original 
+    // begin as the begin/end of the trimmed span
+    int backwardsSeekingCodepoint;
+    while (
+              (end > 0)
+              && end > begin
+              && aIsTrimChar.test(backwardsSeekingCodepoint = text.codePointBefore(end))
+    ) {
+      end -= Character.charCount(backwardsSeekingCodepoint);
+    }
+    
+    // Then, trim at the start
+    int forwardSeekingCodepoint;
+    while (
+              (begin < (text.length() - 1))
+              && begin < end
+              && aIsTrimChar.test(forwardSeekingCodepoint = text.codePointAt(begin))
+    ) {
+      begin += Character.charCount(forwardSeekingCodepoint);
+    }
+      
+    setBegin(begin);
```

Comment:
need to "protect" the update (in case the annotation was in some index in some view);  See Adjusting an existing annotation's begin and end in http://uima.apache.org/doc-uimaj-cookbook.html

Reply:
Is calling `removeFromIndex()` causing extra overhead if the annotation hasn't been even added to the index yet? Is there a way to tell if removing/adding is necessary before doing it?

Reply:
My reply seems to have gotten lost - redoing:  the method suggested in the http://uima.apache.org/doc-uimaj-cookbook.html section on "Adjusting an existing annotation's begin and end" gives a way at the bottom that avoids doing the removeFromIndex etc., if the Feature Structure is not in any index in any view:

my_cas.protectIndexes(() -> {
   // ... arbitrary user code updating "key" features, 
   //     but no checked exceptions are permitted
   //     (because inside a lambda)
   
   a.setBegin(new_value_begin);
   a.setEnd(new_value_end);
   });